### ISSUE TITLE
Building latest RPM

### ISSUE THREAD

**Issue by skynet** [ansible/ansible#12755]
```
git clone git://github.com/ansible/ansible.git --recursive
cd ./ansible
make rpm
sudo rpm -Uvh ./rpm-build/ansible-*.noarch.rpm
```

and it fails:

```
ansible --version
Traceback (most recent call last):
  File "/usr/bin/ansible", line 39, in <module>
    from ansible.errors import AnsibleError, AnsibleOptionsError, AnsibleParserError
ImportError: cannot import name AnsibleOptionsError
```

```
python --version
Python 2.7.10
```

```
/usr/bin/python2.6 -c "from distutils.sysconfig import get_python_lib; import sys; sys.stdout.write(get_python_lib())"
/usr/lib/python2.6/dist-packages
```

```
rpm -q --eval '%{__python}'
/usr/bin/python2.6
```

Looks like it is using Python 2.6

**Comment by chrrrles** [comment]
Hi Skynet - I am unable to reproduce this bug on Fedora Core 21.  What distribution are you using?

**Comment by skynet** [comment]
Using AWS Linux: https://aws.amazon.com/amazon-linux-ami/2015.09-release-notes/

**Comment by kikicarbonell** [comment]
I have the same problem after install ansible 2.0.0.2-1~trusty on Ubuntu 14.04

**Comment by sivel** [comment]
I think @srvg was experiencing some similar issue.  `PYTHONPATH` was being "polluted" with paths from an `easy-install.pth` and it was inserting a path that had a `ansible.errors` file from a different version of Ansible.  

Might be worth looking at the `PYTHONPATH` environment var and looking for any other `errors.py` file.

**Comment by srgvg** [comment]
My problem was another Python package (gandi.cli) inserting paths in `~/.local/lib/python2.7/site-packages/easy-install.pth`:

```
import sys; sys.__plen = len(sys.path)
./gandi.cli-0.16-py2.7.egg
/usr/lib/python2.7/dist-packages
import sys; new = sys.path[sys.__plen:]; del sys.path[sys.__plen:]; p = getattr(sys, '__egginsert', 0); sys.path[p:p] = new; sys.__egginsert = p + len(new)
```

which were at Python run time inserted before the PYTHONPATH stuff.

I got the same error message as in this ticket: it picked up an older (1.9.x) ansible installed system wide.

**Comment by hcavalle** [comment]
I also get the same error, here's my OS version:

```
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:        14.04
Codename:       trusty
```

Full error:

```
Traceback (most recent call last):
  File "/usr/bin//ansible", line 39, in <module>
    from ansible.errors import AnsibleError, AnsibleOptionsError, AnsibleParserError
ImportError: cannot import name AnsibleOptionsError
```

here is tail of `apt-get install ansible` after updating:

```
Get:1 http://archive.ubuntu.com/ubuntu/ trusty/main python-paramiko all 1.10.1-1git1build1 [106 kB]
Get:2 http://ppa.launchpad.net/ansible/ansible/ubuntu/ trusty/main ansible all 2.0.0.2-1ppa~trusty [1,017 kB]
Get:3 http://archive.ubuntu.com/ubuntu/ trusty/universe sshpass amd64 1.05-1 [10.5 kB]
Fetched 1,134 kB in 4s (264 kB/s)                                  
Selecting previously unselected package python-paramiko.
(Reading database ... 491241 files and directories currently installed.)
Preparing to unpack .../python-paramiko_1.10.1-1git1build1_all.deb ...
Unpacking python-paramiko (1.10.1-1git1build1) ...
Selecting previously unselected package sshpass.
Preparing to unpack .../sshpass_1.05-1_amd64.deb ...
Unpacking sshpass (1.05-1) ...
Preparing to unpack .../ansible_2.0.0.2-1ppa~trusty_all.deb ...
Unpacking ansible (2.0.0.2-1ppa~trusty) over (1.5.4+dfsg-1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up python-paramiko (1.10.1-1git1build1) ...
Setting up sshpass (1.05-1) ...
Setting up ansible (2.0.0.2-1ppa~trusty) ...
Installing new version of config file /etc/ansible/hosts ...
Installing new version of config file /etc/ansible/ansible.cfg ...
Processing triggers for python-support (1.0.15) ...
```

**Comment by hcavalle** [comment]
installing from source is actually easier and solves my problem. and you can run dev branch if you want...

**Comment by jsengla** [comment]
I am also experiencing this issue.  It is a non-issue with Fedora or when using an RPM released for FC24, or when using the auto-built RPM on a modern Fedora distro.  However, the RPM auto-built from source and installed on another system can result in this error.
On a pristine copy of RedHat 6.4, the RPM installs the main ansible python executable with a '#!/bin/python' shebang, which breaks when python is just located in /usr/bin/python.  Upon putting a symlink from /bin/python to /usr/bin/python (something present in most, but not all distros), ansible runs and this particular ImportError comes up from the failure to find the site-packages directory.   With the both the auto-built RPM (built on an up-to-date RedHat 6.4 system) and and RPM released for FC24, the ImportError is an issue.
Setting the PYTHONPATH environment variable to include /usr/lib/python2.7/site-packages "fixes" this problem, but the real issue here is that the auto-built RPM installs modules into /usr/lib/python2.7/site-packages even on a system that only has python 2.6.

**Comment by ansibot** [comment]
@skynet closure has been requested for this issue or it has timed out waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jctanner** [comment]
!needs_info

**Comment by jctanner** [comment]
@skynet please indicate if this is still an issue for you.

needs_info

**Comment by ansibot** [comment]
@skynet ping, this issue is still waiting on your feedback. We will close the issue if you do not respond.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@skynet Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibotdev** [comment]
@skynet You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
